src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -18,16 +18,68 @@
 
 #include <lib/core/CHIPPersistentStorageDelegate.h>
 #include <lib/core/DataModelTypes.h>
-#include <lib/core/GroupId.h>
-#include <lib/support/EnforceFormat.h>
-#include <lib/support/logging/Constants.h>
-#include <stdarg.h>
+#include <lib/support/CHIPMemString.h>
+
 #include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 
 namespace chip {
 
+/**
+ * Represents a key used for addressing a specific storage element.
+ *
+ * May contain generic fixed keys (e.g. "g/fidx") or formatted fabric-specific
+ * keys ("f/%x/..." where %x is the fabric index).
+ */
+class StorageKeyName
+{
+public:
+    StorageKeyName(const StorageKeyName & other) = default;
+    StorageKeyName & operator=(const StorageKeyName & other) = default;
+
+    ~StorageKeyName() { memset(mKeyNameBuffer, 0, sizeof(mKeyNameBuffer)); }
+
+    const char * KeyName() const { return mKeyNameBuffer; }
+
+    bool IsInitialized() const { return mKeyNameBuffer[0] != 0; }
+    bool IsUninitialized() const { return mKeyNameBuffer[0] == 0; }
+    bool operator!() const { return IsUninitialized(); }
+
+    operator const char *() const { return mKeyNameBuffer; }
```

Comment:
Suggest removing this, since auto-conversions of classes to `const char *` always are very unexpected.

Reply:
I considered this, but it makes the code very verbose ant it is already long. Like:

```cpp
    ReturnErrorOnFailure(
        storage->SyncSetKeyValue(DefaultStorageKeyAllocator::FabricOpKey(fabricIndex), 
        buf, static_cast<uint16_t>(opKeyLength)));
```

now becomes

```cpp
    ReturnErrorOnFailure(
        storage->SyncSetKeyValue(DefaultStorageKeyAllocator::FabricOpKey(fabricIndex).KeyName(), 
        buf, static_cast<uint16_t>(opKeyLength)));
```

I can do it as a followup if you think is important.